### PR TITLE
Feat: Issue#80 PUT /user/additional_info api endpoints and test cases

### LINKED ISSUE #80: Dev: Update user additional information (PUT /user/additional_info)
## Description
As a student,
I need to create PUT /user/additional_info endpoint and tests,
so that I can give user the ability to update their Personal Details.

## Mocks
NA

### Update [Required]
- [ ] User can update their Personal details profile
- [ ] including test cases

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 day

**Additional Note**
This issue is a part of issue #62 Create/update a profile in BIT

### PR DESCRIPTION
**Pull request by mtreacy002**
### Description
Allow user to update their additional information.

Fixes #80 

### Type of Change:
- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
- tested manually by running the application and update user additional information.
* on successful update:
Before update: 
Screen Shot 2020-07-15 at 9 32 54 pm

After update:
Screen Shot 2020-07-15 at 9 40 36 pm

Success message on Swagger UI:
Screen Shot 2020-07-15 at 11 35 33 pm

* On invalid input (invalid timezone):
Screen Shot 2020-07-15 at 9 15 37 pm

* When user has no existing additional information data in database:
Screen Shot 2020-07-15 at 11 42 57 pm

- by running unittest and all test cases passed
Screen Shot 2020-07-15 at 11 47 46 pm

### Checklist:
- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by mtreacy002** [comment]
Update @ramitsawhney27 and @anitab-org/bridgeintech-maintainers . Here's the PUT /user/additional_info to update user's existing additional information. Can you please provide feedback? Thanks

**Review by ramitsawhney27 — DISMISSED** [review]

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . I've pushed the change requested on Timezone enum as our discussion last night. Now all logic on enum will be dealt by backend, frontend ui will only view enum value.

**Comment by mtreacy002** [comment]
@anitab-org/bridgeintech-maintainers . This PR and all dependent PRs after this one have merge conflict that is caused by the recent merge from [PR#77](https://github.com/anitab-org/bridge-in-tech-backend/pull/77#pullrequestreview-455277155). I will leave the merge conflict in place (not going to solve it for now) until the next PR on the queue ([PR#79](https://github.com/anitab-org/bridge-in-tech-backend/pull/79)) is merged. This way, the merge conflict serves as automatic blocker to this PR so we could prevent accidental merging of this PR before its depenceny/ies (blocker) PR/s get merged. 😉
However, mentors can still reeview and approve the PR with note saying something like "providing merge conflict gets resolved", so I know if the PR is Ok, or if I need to do any modification to it. You'll need to reapprove though once the merge conflict is resolved and force pushed. Thanks

**Comment by isabelcosta** [comment]
@mtreacy002 can you solve this now, that #79 was merged? Ping me when you need my review

**Comment by mtreacy002** [comment]
Update @isabelcosta and @anitab-org/bridgeintech-maintainers . I've solved the merge conflict. Can you please re-review this PR? Ta

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . I've fix the user_id bug mentioned in my [zulip post](https://anitab-org.zulipchat.com/#narrow/stream/237630-bridge-in-tech/topic/Bugs.20.26.20Fixes/near/205030500).
I also placed a temporary solution for bug on issue #94 by set AUTH_COOKIE["user_id"] to == 0 when user send POST /login request. 
Note that setting user_id cookie to 0 only works as soluton in local environment. In dev environment where remote heroku server is used and more than one user access the server at a time, this solution might not work (so, I'll keep issue #94 open as icebox to be revisited after GSoC).


Can you please re-review? Thanks

File: app/api/request_api_utils.py
Comment (isabelcosta) [inline app/api/request_api_utils.py]: why is this 0? @mtreacy002

Comment (mtreacy002) [reply app/api/request_api_utils.py]: @isabelcosta. This is because if the server only just run for the first time, there'll be no AUTH_COOKIE["user_id"] ever initiated in the system. I tried once with `if not AUTH_COOKIE["user_id"]` to check if it's `none` but the system crash since there's no such attribute as `user_id` in the AUTH_COOKIE. But if the system already has someone logged in, their "user_id" need to be cleared out so another logged in person not going to be accidentally associated to it. I also tried to set AUTH_COOKIE["user_id"] = None, but that didn't work as well (forgot what thee error message was).
So here, I used `0` as a default on AUTH_COOKIE["user_id"] since the database will never give an index  of `0` as an ID to an instance (it always starts with 1). So, it's sort of the safest way to instantiate the attribute "user_id" for AUTH_COOKIE but without associating it to any user.

Comment (ramitsawhney27) [reply app/api/request_api_utils.py]: I think we should keep it as None. We can look into the error, it would just be a TypeError, where you can cast None to 0 if needed

File: app/api/dao/user_extension.py
Comment (ramitsawhney27) [inline app/api/dao/user_extension.py]: If not user_id

**Review by ramitsawhney27 — CHANGES_REQUESTED** [review]

File: app/api/request_api_utils.py
Comment (mtreacy002) [reply app/api/request_api_utils.py]: Got it. I changed it to None and use catch the ValueError exception. Perhaps that crash I encountered last time was on something else then 😊

File: app/api/dao/user_extension.py
Comment (mtreacy002) [reply app/api/dao/user_extension.py:90]: Yep, it's now removed completely as I'm applying the EAFP `try catch` 😁
Screen Shot 2020-07-29 at 5 32 21 pm

Comment (mtreacy002) [reply app/api/dao/user_extension.py]: Yep, I removed it completely as now I'm applying EAFP `try catch` instead 😁

Screen Shot 2020-07-29 at 5 32 21 pm

**Comment by mtreacy002** [comment]
Update @ramitsawhney27 and @anitab-org/bridgeintech-maintainers . I've just pushed the requested changes. Can you please re-review? Thanks

File: app/api/dao/user_extension.py
Comment (ramitsawhney27) [reply app/api/dao/user_extension.py]: Very happy to see more usage of EAFP!

**Review by ramitsawhney27 — APPROVED** [review]

**Review by foongminwong — APPROVED** [review]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses were tested as below:

   * _PR 81_  
      _Screenshot/gif/url_:  

      test-pr-81

      _Expected Result_:  As a user, I should be able to update my additional information.
      _Actual Result_: Same as expected

   * _Running PR81 test cases_  
      _Screenshot/gif/url_:  
      
       test-pr-81-tests

      _Expected Result_:  Test cases are run successfully.
      _Actual Result_: Same as expected

3. OS Version: Windows 10

**Pull request merged**
